fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
+@Composable
+private fun HomeButton(
+    onHomeButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onHomeButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_home_24),
+            stringResource(id = R.string.browser_toolbar_home),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun SearchWebButton(
+    onSearchButtonClick: () -> Unit,
+) {
+    IconButton(
+        onClick = { onSearchButtonClick() },
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.mozac_ic_search_24),
+            stringResource(id = R.string.search_hint),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+private fun MenuButton(
+    menuButton: MenuButton,
+) {
+    AndroidView(
+        modifier = Modifier.size(48.dp),
+        factory = { _ -> menuButton },
     )
+}
+
+@Composable
+private fun TabsButton(
+    onTabsButtonClick: () -> Unit,
+    tabCount: Int,
+) {
+    CompositionLocalProvider(LocalContentColor provides FirefoxTheme.colors.iconPrimary) {
+        IconButton(onClick = { onTabsButtonClick() }) {
+            TabCounter(tabCount = tabCount)
+        }
+    }
+}
 
-    val forward = ActionItem(
-        iconId = R.drawable.mozac_ic_forward_24,
-        descriptionResourceId = R.string.browser_menu_forward,
+@Composable
+private fun HomeNavBarPreviewRoot(isPrivateMode: Boolean) {
+    val context = LocalContext.current
+    // private mode preview keeps using wrong color
+    val colorId = if (isPrivateMode) {
+        ThemeManager.resolveAttribute(R.attr.textOnColorPrimary, context)
+    } else {
+        ThemeManager.resolveAttribute(R.attr.textPrimary, context)
+    }
+    val menuButton = MenuButton(context).apply {
+        setColorFilter(
+            ContextCompat.getColor(
+                context,
+                colorId,
+            ),
+        )
+    }
+
+    HomeNavBar(
+        onSearchButtonClick = {},
+        menuButton = menuButton,
+        onTabsButtonClick = {},
+        isPrivateMode = false,
+        browserStore = BrowserStore(),
     )
+}
 
-    val tabs = ActionItem(
-        iconId = R.drawable.mozac_ui_tabcounter_box,
-        descriptionResourceId = R.string.mozac_tab_counter_content_description,
-        type = ItemType.TAB_COUNTER,
+@Composable
+private fun OpenTabNavBarNavBarPreviewRoot(isPrivateMode: Boolean) {
+    val context = LocalContext.current
+    // private mode preview keeps using wrong color
+    val colorId = if (isPrivateMode) {
+        ThemeManager.resolveAttribute(R.attr.textOnColorPrimary, context)
+    } else {
+        ThemeManager.resolveAttribute(R.attr.textPrimary, context)
+    }
+    val menuButton = MenuButton(context).apply {
+        setColorFilter(
+            ContextCompat.getColor(
+                context,
+                colorId,
+            ),
+        )
+    }
```

Comment:
if we're able to ditch `MenuButton` in favor of an `IconButton` + click lambda, we shouldn't need this logic block

Reply:
sadly, it looks like we have to keep the menu button for time being